Source/Core/UICommon/GameFile.cpp

```diff
@@ -163,6 +165,34 @@ GameFile::GameFile(std::string path) : m_file_path(std::move(path))
     m_platform = DiscIO::Platform::ELFOrDOL;
     m_blob_type = DiscIO::BlobType::DIRECTORY;
   }
+
+  if (!IsValid() && GetExtension() == ".json")
+  {
+    auto descriptor = DiscIO::ParseGameModDescriptorFile(m_file_path);
+    if (descriptor)
+    {
+      GameFile proxy(descriptor->base_file);
+      if (proxy.IsValid())
+      {
+        m_valid = true;
+        m_file_size = File::GetSize(m_file_path);
+        m_long_names.emplace(DiscIO::Language::English, std::move(descriptor->display_name));
+        m_internal_name = proxy.GetInternalName();
+        m_game_id = proxy.GetGameID();
+        m_gametdb_id = proxy.GetGameTDBID();
+        m_title_id = proxy.GetTitleID();
+        m_maker_id = proxy.GetMakerID();
+        m_region = proxy.GetRegion();
+        m_country = proxy.GetCountry();
+        m_platform = proxy.GetPlatform();
+        m_revision = proxy.GetRevision();
+        m_disc_number = proxy.GetDiscNumber();
+        if (ReadPNGBanner(descriptor->banner))
+          m_volume_banner = std::move(m_pending.custom_banner);
```

Comment:
I would suggest putting the `ReadPNGBanner` call in `CustomBannerChanged` instead of the constructor so that the game list will finish populating quicker. (`CustomBannerChanged` is called by `UpdateAdditionalMetadata` after all GameFiles have been constructed and inserted into the game list. Perhaps its name doesn't do a great job of explaining what it does...)